cookbooks/bcpc-hadoop/recipes/namenode_master.rb

```diff
@@ -101,16 +109,55 @@
 # So we fake the formatting of the txn directories by copying over current/VERSION
 # this tricks the journalnodes and namenodes into thinking they've been formatted.
 
-ruby_block "grab the format UUID File" do
+ruby_block "create-format-UUID-File" do
   block do
     Dir.chdir("/disk/#{node[:bcpc][:hadoop][:mounts][0]}/dfs/") do
       system("tar czvf #{Chef::Config[:file_cache_path]}/nn_fmt.tgz nn/current/VERSION jn/#{node.chef_environment}/current/VERSION")
     end
-    make_config("namenode_txn_fmt", Base64.encode64(IO.read("#{Chef::Config[:file_cache_path]}/nn_fmt.tgz")));
   end
-  action :nothing
-  subscribes :run, "service[generally run hadoop-hdfs-namenode]", :immediately
-  only_if { File.exists?("/disk/#{node[:bcpc][:hadoop][:mounts][0]}/dfs/nn/current/VERSION") }
+  action :run
+  only_if { File.exists?("/disk/#{node[:bcpc][:hadoop][:mounts][0]}/dfs/nn/current/VERSION") and  File.exists?("/disk/#{node[:bcpc][:hadoop][:mounts][0]}/dfs/jn/#{node.chef_environment}/current/VERSION") }
+end
+
+ruby_block "upload-format-UUID-File" do
+  block do
+    cmdStrCount = "zgrep -a -i layoutVersion #{Chef::Config[:file_cache_path]}/nn_fmt.tgz|wc -l"
+    cmdStrUnqCount = "zgrep -a -i layoutVersion #{Chef::Config[:file_cache_path]}/nn_fmt.tgz|uniq|wc -l"
+    cmdStrLayVersn = "zgrep -a -i layoutVersion #{Chef::Config[:file_cache_path]}/nn_fmt.tgz|uniq|cut -d'=' -f2"
+
+    cmd = Mixlib::ShellOut.new(cmdStrCount, :timeout => 10).run_command
+    cmd.error!
+    Chef::Log.debug("Total number of version lines : #{cmd.stdout}") 
+    if cmd.stdout.to_i != 2
+      Chef::Log.fatal("Couldn't find required number of layoutVersion records");
```

Comment:
After every `Chef::Log.fatal` we should `raise` to stop processing this block; perhaps set the resource attribute `ignore_failure` so we continue on but at least the problem will be logged.